### PR TITLE
[nrf5] Enable enhancement features of OpenThread

### PR DESCRIPTION
**Pull request by bukepo**
- Enable jam detection
- Enable child supervisoin

### CODE REVIEW

File: build/nrf5/nrf5-openthread.mk [hunk 1]
```diff
@@ -161,7 +161,9 @@ OPENTHREAD_CONFIGURE_OPTIONS = \
     --srcdir="$(OPENTHREAD_ROOT)" \
     --enable-ftd \
     --enable-mtd \
+    --enable-child-supervision \
     --enable-diag \
+    --enable-jam-detector \
```
Comment (jwhui) [inline build/nrf5/nrf5-openthread.mk]:
```suggestion
    --enable-jam-detection \
```

**Comment by jaylogue** [comment]
Why do we want these on for all projects that use OpenWeave?  Are these features necessary/appropriate for all use cases?  What if a project doesn't want these features enabled?

**Comment by bukepo** [comment]
@jaylogue I think these features are appropriate for the lock and border router. Since the current makefile assumes an old OpenThread commit, it doesn’t support enable/disable all features in OpenThreadConfig.h. So it seems we have to enable it here for now before we move to new OpenThread.

**Comment by jaylogue** [comment]
@bukepo This makefile is potentially used by any product that uses OpenWeave on the nRF52840.  So the default OpenThread options we select should be the right ones for all such products, not just locks.

I'm fine with this change if we think it should apply universally.  However, if we don't, then I think individual products should enable them as needed in their product makefiles by overriding OPENTHREAD_CONFIGURE_OPTIONS.  

For example, you can enable jam detection and child supervision in the [openweave-nrf52840-lock-example](https://github.com/openweave/openweave-nrf52840-lock-example) by adding the following lines to the top-level [Makefile](https://github.com/openweave/openweave-nrf52840-lock-example/blob/master/Makefile):

```
OPENTHREAD_CONFIGURE_OPTIONS += \
    --enable-child-supervision \
    --enable-jam-detection
```

(Insert these lines after the include statements and before the call to GenerateBuildRules).

This would also be a quick way for our device partners to test this feature without waiting for an OpenWeave change.

**Comment by bukepo** [comment]
Moved into lock example.

**Pull request closed without merging**